src/qurveros/spacecurve.py

```diff
@@ -249,20 +249,41 @@ def _frenet_dict_fun(x, params):
 
         self._frenet_dict_fun = _frenet_dict_fun
 
-    def set_params(self, params):
-
-        """
-        Sets the default parameters of the curve.
-        The associated elements are set to None so that the geometric
-        quantities, the robustness properties and the control fields are
-        evaluated for the new set of auxiliary parameters.
-        """
-
-        self.params = params
+def _cast_params_to_float(self, params):
```

Comment:
Since this function is only used once now (in `set_params`), could you please move all of the logic to `set_params` and remove this function? This will simplify the class and eliminate the overhead of the extra function call.